client/app/containers/test-run-info/test-run-info.controller.js

```diff
@@ -680,20 +698,59 @@ const testRunInfoController = function testRunInfoController(
 
     $scope.logs = [];
     var unrecognizedImages = {};
+    const screenshotExtension = '.png';
 
     function collectLogs(log) {
-        switch (log.level) {
-            case 'META_INFO':
-                collectScreenshots(log);
-                break;
-            default:
-                $scope.logs.push(log);
-                break;
+        if (log.level === 'META_INFO') {
+            collectScreenshots(log);
+        } else if (log.kind === 'screenshot') {
+            collectScreenshotsForNewAgent(log);
+        } else {
+            $scope.logs.push(log);
         }
         vm.filteredLogs = $scope.logs;
         $scope.$applyAsync();
     };
 
+    function collectScreenshotsForNewAgent(log) {
+        const logToAttache = $scope.logs.find((l, index) => {
```

Comment:
use full name instead of `l`

Reply:
log variable name is already defined in the scope